### ISSUE TITLE
Bug: Missing Sanity Check in convert_hf_to_gguf.py

### ISSUE THREAD

**Issue by mesibo** [ggerganov/llama.cpp#9245]
### What happened?

The script convert_hf_to_gguf.py does not perform a sanity check on the completeness of the model files. We encountered an issue where one of the models we downloaded was incomplete (specifically, a safesensors file was missing). Despite this, convert_hf_to_gguf.py proceeded with the conversion without generating any errors or warnings.

Expected Behavior: The script should validate that all required files are present before converting. Since Hugging Face downloads are organized with sequence and total files (e.g., model-00001-of-00002), it should be easy to validate. 

Steps to Reproduce:

1. Download a model from HF and delete one of the safesensors file.
2. Run convert_hf_to_gguf.py on the incomplete set.
3. the script completes without any indication of missing files.

### Name and Version

version: 3631 (7d787ed9)
built with cc (GCC) 8.5.0 20210514 (Red Hat 8.5.0-22) for x86_64-redhat-linux

### What operating system are you seeing the problem on?

Linux

**Comment by compilade** [comment]
> Steps to Reproduce:
> 
>   1. Download a model from HF and delete one of the safesensors file.
>   2. Run convert_hf_to_gguf.py on the incomplete set.
>   3. the script completes without any indication of missing files.

@mesibo

I tried, but I can't reproduce not getting an error. (tried on version <https://github.com/ggerganov/llama.cpp/commit/8f1d81a0b6f50b9bad72db0b6fcd299ad9ecd48c>, but the following has been in `convert_hf_to_gguf.py` since #7075)

I get

```
ValueError: Mismatch between weight map and model parts for tensor names:
```

and then a big list of missing tensors.

This comes from

https://github.com/ggerganov/llama.cpp/blob/8f1d81a0b6f50b9bad72db0b6fcd299ad9ecd48c/convert_hf_to_gguf.py#L174-L176

**Comment by compilade** [comment]
@mesibo

I just now realized that you're right, when there's only one `.safetensors` detected there is no sanity check, even when there still is a `model.safetensors.index.json`.

So what you're describing can be reproduced by having only a single `.safetensors` of the model when it should have been multi-part. In my previous test I used a model with 4 parts and removed one.

This could be improved by cross-checking the index files when present even when only a single model file is detected.
